src/xdocs/cmdline.xml.vm

```diff
@@ -204,6 +204,20 @@ java -D&lt;property&gt;=&lt;value&gt;  \
           &#xa0;&#xa0;&#xa0;&#xa0;MyClass.java
         </source>
       </div>
+      <p>
+        To run Checkstyle directly from the JAR file using
+        the <code>-jar</code> option when the JAR file and test file are not in same folder
+        or if CLI is launched from any other path. Download latest
+        <a href="https://github.com/checkstyle/checkstyle/releases/download/checkstyle-${projectVersion}/checkstyle-${projectVersion}-all.jar">
```

Comment:
I don't see the point of repeating the download location and use the latest 3 times in the same document. It seems very repetitive. I also disagree that we should be linking to a direct download location as we have a general location at https://checkstyle.org/#Download for all downloads and just specify only GH releases has the all jar.

> The problem here almost in every case was the download path and cli launch path being different (or) the test file/config not located at the same directory as jar.

I also feel none of these updates in this PR really explain what seems to be the confusion, but I have not seen these conversations. It seems to me we would have some wording that says all paths are relative to the current working directory but really this should be understood when working with command lines.

Also, this confusion seems to possibly point to some issue with error messages checkstyle prints. Are they not getting the hint the files don't exist (wrong path) or are we not displaying proper error messages?

In the end, I think this documentation needs some clean up and the other things I mentioned looked at.

Reply:
> Also, this confusion seems to possibly point to some issue with error messages checkstyle prints. Are they not getting the hint the files don't exist (wrong path) or are we not displaying proper error messages?

```
E:\MANISH\Workspace>cd checkstyle

E:\MANISH\Workspace\checkstyle>java  -jar checkstyle-10.14.2-all.jar -c config.xml Test.java
Error: Unable to access jarfile checkstyle-10.14.2-all.jar

E:\MANISH\Workspace\checkstyle>cd ..

E:\MANISH\Workspace>ls
CheckstyleCMD.java     Test.java                       checkstyle                  checkstyle-10.14.2-all.jar  config.xml
MANISH-K-07.github.io  apache-groovy-binary-2.4.8.zip  checkstyle-10.13.0-all.jar  checkstyle-contribution     groovy-2.4.8

E:\MANISH\Workspace>java  -jar checkstyle-10.14.2-all.jar -c config.xml NotTest.java
Files to process must be specified, found 0.

```
@rnveach , The error messages we echo actually suffice if the user understands cli, this however is not the case for all our contributors..

> but I have not seen these conversations.

We keep getting posts with error log asking why the jar isn't running even when the cmd is copied exactly from docs.
[An example convo](https://matrix.to/#/!TnyOUUMSLLLwazyDHl:gitter.im/$8xysqsMoW4W20LyaJbwjIqO4G_TltjAjVmiXYtlUEzE?via=gitter.im&via=matrix.org&via=mozilla.org)

The following can be done if required....
```
E:\MANISH\Workspace\checkstyle>java  -jar checkstyle-10.14.2-all.jar -c config.xml Test.java
Error: Unable to access jarfile checkstyle-10.14.2-all.jar in the current working directory.

E:\MANISH\Workspace>java  -jar checkstyle-10.14.2-all.jar -c config.xml NotTest.java
Files to process must be specified, found 0 in the current working directory.

```

Reply:
Unless I am missing something, using project properties here will never work since that version has not been released yet.

Reply:
My limited understanding is it uses the version in the POM. If our POM says snapshot, it will say snapshot and snapshot is what is creating during package/install. We create this documentation as part of our release process, without snapshot, and it looks correct. See https://checkstyle.org/cmdline.html#Download_and_Run

Reply:
> My limited understanding is it uses the version in the POM. If our POM says snapshot, it will say snapshot and snapshot is what is creating during package/install. We create this documentation as part of our release process, without snapshot, and it looks correct. See https://checkstyle.org/cmdline.html#Download_and_Run

Ah, right on, forgot that we do change this value during release